aiidalab_widgets_base/computational_resources.py

```diff
@@ -197,29 +197,21 @@ def refresh(self, _=None):
 
     @traitlets.validate("value")
     def _validate_value(self, change):
-        """If code is provided, set it as it is. If code's label is provided,
-        select the code and set it."""
-        code = change["value"]
+        """Check if the code is valid in DB"""
+        code_uuid = change["value"]
         self.output.value = ""
 
         # If code None, set value to None.
-        if code is None:
+        if code_uuid is None:
             return None
 
-        if isinstance(code, str):  # Check code by label.
-            if code in self.codes:
-                return self.codes[code]
-            self.output.value = f"""No code named '<span style="color:red">{code}</span>'
-            found in the AiiDA database."""
-        elif isinstance(code, orm.Code):  # Check code by value.
-            label = self._full_code_label(code)
-            if label in self.codes:
-                return code
-            self.output.value = f"""The code instance '<span style="color:red">{code}</span>'
-            supplied was not found in the AiiDA database."""
-
-        # This place will never be reached, because the trait's type is checked.
-        return None
+        try:
```

Comment:
I think for code we still want to support selecting code by their labels, no?
In which case this validation should not be here, because code labels will not be valid UUIDs? Or are they?

Reply:
> I think for code we still want to support selecting code by their labels, no?

No, I don't plan to support the label as value of the widget. What do you think? I'll update the docstring if you agree with it. 

Also pinning @yakutovicha for comment, since you author this widget.

Reply:
Originally, when developing this, I aimed to mimic the behaviour of `verdi code show` (or similar) that would accept any way of selecting the code (by `name`, `id`, `uuid`). It is nice to have, but if using only `uuid` lets us significantly simplify the code, maybe let's stick to `uuid` only.

I do not have strong preferences w.r.t both options.

Reply:
I guess we are all only back next week 😄 . I am off till Tuesday. @unkcpz mentioned that he is packed till the end of this week also.

Reply:
> think for codes and computers it makes a lot of sense to be able to use the label because that's what is independent of the concrete database.

I believe that is not exactly true. Suppose a computer was exported to a database which already contains a computer with the same label. The only way they could be distinguished is by UUID. So I am not sure what is the advantage of keeping the label here.